### ISSUE TITLE
rule conditions don't apply if the field is entirely cleared

### ISSUE THREAD

**Issue by skairunner** [eclipsesource/jsonforms#2396]
### Describe the bug

Rules don't evaluate as false if the property doesn't exist.

### Expected behavior

I expect that a missing property would evaluate as false for the purposes of a rule. Alternatively, I would expect some property on the rules object that can let me control whether a missing property is true or false.

### Steps to reproduce the issue

Here is a code sandbox that demonstrates the issue: https://codesandbox.io/p/sandbox/jsonforms-wp8k83

The relevant part of the UI schema:
```js
{
elements: [
      {
        type: "Control",
        scope: "#/properties/name",
        options: { multi: false }
      },
      {
        type: "Control",
        scope: "#/properties/enum",
        rule: {
          effect: "ENABLE",
          condition: {
            scope: "#/properties/name",
            schema: { minLength: 2 }
          },
        },
      }
    ]
}
```

We have a `name` field, and then we have an `enum` field which has a rule "if `name` is at least two characters, enable". 

- When the page is first loaded, the enum dropdown is disabled.
- Type two characters into the name field. The enum dropdown enables itself as expected.
- Delete one character from the name field. The enum dropdown disables itself as expected.
- Delete all characters from the name field. The enum dropdown re-enables itself.

If you log the generated json data, when you delete the `name` field's contents it removes the property entirely from the object rather than leaving it as a blank string. It seems that the rule condition falls through when the field is missing.

### Which Version of JSON Forms are you using?

v3.3.0

### Package

Core, React Bindings, React Vanilla Renderers

**Comment by sdirix** [comment]
We already have such an option, it is called `failWhenUndefined`, see [here](https://github.com/eclipsesource/jsonforms/commit/8fc90d3ee6221cd0e719557853a81f6609dd7d43).

So you can use:

```
        rule: {
          effect: "ENABLE",
          condition: {
            scope: "#/properties/name",
            schema: { minLength: 2 },
            failWhenUndefined: true
          },
        },
```

Please use our [community forum](https://jsonforms.discourse.group/t/disable-fields-based-on-empty-field/2327) for questions.

**Comment by skairunner** [comment]
It seems a bit unfortunate that this schema isn't mentioned anywhere in the documentation aside from the automatically generated ones: https://www.google.com/search?q=site%3Ajsonforms.io+failWhenUndefined

This page seems like a good place to mention it: https://jsonforms.io/docs/uischema/rules

**Comment by sdirix** [comment]
Good point! Would you like to add this to the [website](https://github.com/eclipsesource/jsonforms2-website/blob/c25a6bbb80986042934826eb5939bf8920cf770a/content/docs/uischema/rules.mdx)?
